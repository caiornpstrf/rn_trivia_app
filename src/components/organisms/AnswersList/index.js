/**
 * @summary list of answers of a given Question
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 06:35:17
 * Last modified  : 2021-04-12 06:36:03
 */

import React, {useState} from 'react';
import {StyleSheet, ScrollView} from 'react-native';
import {array, func} from 'prop-types';

import {RadioCard} from '_components/molecules';

const AnswersList = ({data, onItemSelected}) => {
  const [selected, setSelected] = useState('');
  return (
    <ScrollView contentContainerStyle={styles.container}>
      {data.map((item, index) => (
        <RadioCard
          key={`answer_${index}`}
          description={item}
          onPress={() => {
            onItemSelected({item, index});
            setSelected(item);
          }}
          isSelected={item === selected}
          style={styles.card}
        />
      ))}
    </ScrollView>
  );
};

AnswersList.propTypes = {
  data: array.isRequired,
  onItemSelected: func,
};

AnswersList.defaultProps = {
  onItemSelected: ({item, index}) => console.log(`Selected index: ${index}`),
};

export default AnswersList;

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
  card: {
    marginBottom: 20,
  },
});
