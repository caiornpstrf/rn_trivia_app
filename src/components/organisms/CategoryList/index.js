/**
 * @summary List of Buttons used to select a category
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 06:00:13
 * Last modified  : 2021-04-12 06:35:10
 */

import React from 'react';
import {StyleSheet, ScrollView} from 'react-native';
import {array, func} from 'prop-types';

import {Button} from '_components/atoms';

const CategoryList = ({data, onItemSelected}) => {
  return (
    <ScrollView contentContainerStyle={styles.container}>
      {data.map((item, index) => (
        <Button
          key={`category_${index}`}
          label={item}
          onPress={() => onItemSelected({item, index})}
          style={styles.card}
        />
      ))}
    </ScrollView>
  );
};

CategoryList.propTypes = {
  data: array.isRequired,
  onItemSelected: func,
};

CategoryList.defaultProps = {
  onItemSelected: ({item, index}) => console.log(`Selected index: ${index}`),
};

export default CategoryList;

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
  card: {
    marginBottom: 15,
  },
});
