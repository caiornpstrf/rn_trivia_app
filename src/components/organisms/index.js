export {default as AnswersList} from './AnswersList';
export {default as CategoryList} from './CategoryList';
