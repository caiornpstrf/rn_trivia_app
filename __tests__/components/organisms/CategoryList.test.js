import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {CategoryList} from '_components/organisms';

const data = [
  {id: 0, name: 'Category 0'},
  {id: 1, name: 'Category 1'},
  {id: 2, name: 'Category 2'},
  {id: 3, name: 'Category 3'},
];

it('renders correctly', () => {
  const {toJSON} = render(<CategoryList data={data} />);
  expect(toJSON()).toMatchSnapshot();
});

it('presses all the buttons successfully', () => {
  const mockFn = jest.fn();
  const {getByText} = render(
    <CategoryList data={data} onItemSelected={mockFn} />,
  );
  data.forEach(item => {
    fireEvent.press(getByText(item.name));
  });
  expect(mockFn).toBeCalledTimes(data.length);
});
