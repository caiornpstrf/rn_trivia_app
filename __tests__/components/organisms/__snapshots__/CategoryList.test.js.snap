// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7F7",
            "borderColor": "#78809A",
            "borderRadius": 10,
            "borderWidth": 1,
            "elevation": 4,
            "height": 45,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
          Object {
            "marginBottom": 15,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#78809A",
            "fontSize": 33,
            "fontWeight": "bold",
          }
        }
      >
        Category 0
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7F7",
            "borderColor": "#78809A",
            "borderRadius": 10,
            "borderWidth": 1,
            "elevation": 4,
            "height": 45,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
          Object {
            "marginBottom": 15,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#78809A",
            "fontSize": 33,
            "fontWeight": "bold",
          }
        }
      >
        Category 1
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7F7",
            "borderColor": "#78809A",
            "borderRadius": 10,
            "borderWidth": 1,
            "elevation": 4,
            "height": 45,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
          Object {
            "marginBottom": 15,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#78809A",
            "fontSize": 33,
            "fontWeight": "bold",
          }
        }
      >
        Category 2
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7F7",
            "borderColor": "#78809A",
            "borderRadius": 10,
            "borderWidth": 1,
            "elevation": 4,
            "height": 45,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
          Object {
            "marginBottom": 15,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#78809A",
            "fontSize": 33,
            "fontWeight": "bold",
          }
        }
      >
        Category 3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
